tests/rptest/tests/shadow_indexing_firewall_test.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Redpanda Data, Inc.
+#
+# Licensed as a Redpanda Enterprise file under the Redpanda Community
+# License (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import SISettings
+
+from rptest.clients.types import TopicSpec
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.util import (
+    produce_until_segments,
+    wait_for_segments_removal,
+    firewall_blocked,
+)
+
+# Log errors expected when connectivity between redpanda and the S3
+# backend is disrupted
+CONNECTION_ERROR_LOGS = [
+    "archival - .*Failed to create archivers",
+
+    # e.g. archival - [fiber1] - service.cc:484 - Failed to upload 3 segments out of 4
+    r"archival - .*Failed to upload \d+ segments"
+]
+
+
+class ShadowIndexingFirewallTest(RedpandaTest):
+    log_segment_size = 1048576  # 1MB
+    retention_bytes = 1024  # 1 KB
+    log_compaction_interval_ms = 1000
+
+    s3_topic_name = "panda-topic"
+    topics = (TopicSpec(name=s3_topic_name,
+                        partition_count=1,
+                        replication_factor=3), )
+
+    def __init__(self, test_context):
+        si_settings = SISettings(cloud_storage_reconciliation_interval_ms=500,
+                                 cloud_storage_max_connections=5,
+                                 log_segment_size=self.log_segment_size)
+        self.s3_bucket_name = si_settings.cloud_storage_bucket
+
+        extra_rp_conf = dict(
+            log_compaction_interval_ms=self.log_compaction_interval_ms,
+            retention_bytes=self.retention_bytes,
+            cloud_storage_enable_remote_read=True,
+            cloud_storage_enable_remote_write=True)
+
+        super(ShadowIndexingFirewallTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=extra_rp_conf,
+                             si_settings=si_settings)
+
+        self._s3_port = si_settings.cloud_storage_api_endpoint_port
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.rpk = RpkTool(self.redpanda)
+
+    def setUp(self):
+        super().setUp()  # topic is created here
+
+    def tearDown(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
```

Comment:
Shouldn't have to empty bucket here since that get's done automatically in RedpandaService teardown

https://github.com/redpanda-data/redpanda/blob/7f78673bb6ddb73708794e3195ab5453e3e9d558/tests/rptest/services/redpanda.py#L1068

Reply:
then I should call `redpanda.clean_node`, right? Cause I think now it's not called anywhere in the code

Reply:
No need. Ducktape automatically calls clean_node and stop_node on all services for us.

See the ducktape service registry https://github.com/confluentinc/ducktape/blob/fb6a86acc2e11653f464a2e5111c860a436256ea/ducktape/services/service_registry.py#L60

And the ducktape RunnerClient, the thing that runs a test, will automatically call the necessary service registry methods
https://github.com/confluentinc/ducktape/blob/fb6a86acc2e11653f464a2e5111c860a436256ea/ducktape/tests/runner_client.py#L253

Reply:
Fixed